Compatibility with Laravel 5

rafelsanso opened the pull request.

rafelsanso committed:
Update README.md

Updating readme

rafelsanso committed:
- Change filters to middleware
 - Update tests
 - Prevent error if not goals are defined

rafelsanso committed:
- Ending integration with Laravel 5
- Catching exception when trying to use an experiment without run the install command
- Moving config file with correct command

rafelsanso committed:
Update README.md

Add new install options with Laravel 5

rafelsanso committed:
Update README.md

rafelsanso committed:
- Testing Travis

garygreen commented:
There's a number of issues with this PR, like the console commands, tests. Also I don't think @jenssegers would like to remove support for L4. I've already got both L4 and L5 working in pr #7, it's just the tests that need updating (will try and update soon).

jenssegers commented:
I do prefer to keep support for both L4 and L5. But if it is too hard we could tag a new major release.

jenssegers commented:
@rafelsanso @garygreen I have been looking into the PR's last night. At first I wanted to keep supporting 4.x, 5.0 and 5.1. But after writing some code I kind of want to go for a new major release specifically for Laravel 5.1 LTS. I think keeping support for all 3 Laravel versions will make the codebase unnecessary complicated. I also have some new idea's that I want to integrate.

I can make separate branches with support for older versions, but they might not have all the features of the master branch.

garygreen commented:
@jenssegers sounds good to me. You could use my PR as a base and strip out some L5 specific stuff (mainly the `Helpers`). I think using `Config` stuff inside `Tester.php` is bad practise, so it would be useful to bootstrap those in the service provider, as I have done in my PR. It makes tests easier as well.

rafelsanso committed:
- Removing compatibility with Laravel 4 in composer.json

rafelsanso committed:
Merge branch 'master' of https://github.com/rafelsanso/laravel-ab

jenssegers commented:
Guys, I want your opinion. Do you think it would be a good idea to integrate Google Analytics for stats tracking?

rafelsanso commented:
Hi @jenssegers!

I think it's good idea. 
In fact, I use for statistical analytics. Style:

```
@if (AB::experiment('text1'))
        ga('send', 'event', 'experiments', 'homepage', 'text1');
@elseif (AB::experiment('text2'))
        ga('send', 'event', 'experiments', 'homepage', 'text2');
@endif
```

The advantage is we can obtain more data from Analytics visits, like the bounce rate and the number of page views.

And another thing that I see very interesting is save goals by javascript events. In my case I'm register the goal by pressing a button, not when you switch views.

garygreen commented:
Makes sense to me. I assume it will be optional?

jenssegers commented:
I had a lot of issues integrating it with GA @rafelsanso. Can you show me a screenshot of what the reporting looks like with the `ga('send', 'event', 'experiments', 'homepage', 'text1');`. You can always send it to my email address if you do want to put it public :)

rafelsanso commented:
Of course!
Now I prepare an email with the example that I have. Can you tell me your email?

jenssegers commented:
You can send it to hello at jenssegers.be

rafelsanso commented:
I can't see the "hello" menú :/

jenssegers commented:
Replace the at with @ and remove the whitespaces :p

rafelsanso commented:
jajaja, sorry xD

rafelsanso commented:
You've got the mail :)

rafelsanso commented:
Some progress of the project?

rafelsanso committed:
Update with new functions

- Update to Laravel 5.1
- Add method hasExperiments
- Add method currentExperiment
- Fix issue with config files
- Fix command for make report

rafelsanso committed:
Update Readme

rafelsanso committed:
Update PHP version from travis

rafelsanso committed:
Update readme

rafelsanso committed:
Update phpunit

bart commented:
@jenssegers @rafelsanso Any updates on this?

isbkch commented:
Any updates on this ?

ben182 commented:
I published a new [package ](https://github.com/ben182/laravel-ab) for ab testing that supports laravel 5.
